spec/support/spec_helpers/user_spec_helper.rb

```diff
@@ -2,8 +2,10 @@ module UserSpecHelper
   # @param user [User] User to log in
   # @param referrer [String] (Optional) path to which user should be redirected after signing in.
   def sign_in_user(user, referrer: nil)
-    user.regenerate_login_token if user.login_token.blank?
-    sign_in_path = user_token_path(token: user.login_token, referrer: referrer)
+    user.regenerate_login_token
```

Comment:
It will be useful for this function (and other functions like this) to accept an optional `update_timestamp: false (default)` keyword argument.

That way the second DB update query (below) can be avoided. I see that these two steps are performed in series in a bunch of places.